File: web/src/api/shop.ts
import client, { unwrap, type PageData } from './client'

export type ShopPlatform = 'doudian' | 'taobao' | 'pinduoduo'
export type PluginStatus = 'unbound' | 'online' | 'offline'

export interface MarketplaceShop {
  id: number
  name: string
  platform: ShopPlatform
  platformLabel: string
  bindCode: string
  pluginStatus: PluginStatus
  pluginAvailable: boolean
  platformShopId?: string
  platformShopName?: string
  lastSyncAt?: string
  lastSeenAt?: string
  nextSyncAt?: string
  syncRequested?: boolean
  pendingTicketCount?: number
  remark?: string
  createdAt: string
  updatedAt: string
}

export interface FilterCard {
  groupName: string
  cardKey: string
  cardLabel: string
  count: number
  sortOrder: number
}

export interface AftersaleTicket {
  id: number
  platformAftersaleId: string
  orderNo: string
  productTitle: string
  productImage?: string
  sku: string
  productTags?: string
  tags?: string
  qty: number
  buyQty: number
  payAmount: string
  refundAmount: string
  aftersaleType: string
  reason: string
  status: string
  timeoutText?: string
  timeoutAction?: string
  deadlineAt?: string
  remainSeconds?: number
  dispute?: string
  logistics?: string
  logisticsBuyerStatus?: string
  logisticsShipStatus?: string
  needIntercept?: boolean
  returnLogisticsNo?: string
  shipLogisticsNo?: string
  applyTime?: string
  cardKeys: string[]
  syncedAt: string
}

export interface ShopWorkbench {
  shop: MarketplaceShop
  cards: FilterCard[]
  tickets: AftersaleTicket[]
  total: number
  page: number
  pageSize: number
  lastSyncAt?: string
}

export const PLATFORM_OPTIONS: { value: ShopPlatform; label: string }[] = [
  { value: 'doudian', label: '抖店' },
  { value: 'taobao', label: '淘宝' },
  { value: 'pinduoduo', label: '拼多多' },
]

export const PLUGIN_STATUS_MAP: Record<PluginStatus, { label: string; type: '' | 'success' | 'warning' | 'info' | 'danger' }> = {
  unbound: { label: '未绑定', type: 'info' },
  online: { label: '在线', type: 'success' },
  offline: { label: '离线', type: 'danger' },
}

export interface PluginSetting {
  pluginSyncIntervalMin: number
}

export const PLUGIN_SYNC_OPTIONS: { value: number; label: string }[] = [
  { value: 15, label: '每 15 分钟' },
  { value: 30, label: '每 30 分钟' },
  { value: 60, label: '每 1 小时' },
  { value: 120, label: '每 2 小时' },
  { value: 360, label: '每 6 小时' },
  { value: 720, label: '每 12 小时' },
  { value: 1440, label: '每 24 小时' },
]

export async function fetchPluginSetting() {
  return unwrap<PluginSetting>(await client.get('/plugin-settings'))
}

export async function savePluginSetting(data: PluginSetting) {
  return unwrap<PluginSetting>(await client.put('/plugin-settings', data))
}

export async function fetchShops() {
  return unwrap<MarketplaceShop[]>(await client.get('/shops'))
}

export async function fetchShop(id: number) {
  return unwrap<MarketplaceShop>(await client.get(`/shops/${id}`))
}

export async function createShop(data: { name: string; platform?: ShopPlatform; remark?: string }) {
  return unwrap<MarketplaceShop>(await client.post('/shops', data))
}

export async function updateShop(id: number, data: { name?: string; remark?: string }) {
  return unwrap<MarketplaceShop>(await client.put(`/shops/${id}`, data))
}

export async function deleteShop(id: number) {
  return unwrap<{ deleted: boolean }>(await client.delete(`/shops/${id}`))
}

export async function resetShopBind(id: number) {
  return unwrap<MarketplaceShop>(await client.post(`/shops/${id}/reset-bind`))
}

export async function requestShopSync(id: number) {
  return unwrap<MarketplaceShop>(await client.post(`/shops/${id}/request-sync`))
}

export async function fetchShopWorkbench(id: number, params?: {
  cardKey?: string
  keyword?: string
  page?: number
  pageSize?: number
}) {
  return unwrap<ShopWorkbench>(await client.get(`/shops/${id}/workbench`, { params }))
}

export async function fetchShopTickets(id: number, params?: {
  cardKey?: string
  keyword?: string
  page?: number
  pageSize?: number
}) {
  return unwrap<PageData<AftersaleTicket>>(await client.get(`/shops/${id}/tickets`, { params }))
}

export interface ReturnPackage {
  id: number
  shopId: number
  shopName: string
  platformAftersaleId: string
  orderNo: string
  productTitle: string
  productImage?: string
  sku: string
  qty: number
  buyQty?: number
  payAmount: string
  refundAmount: string
  aftersaleType: string
  reason: string
  status: string
  orderInfo?: string
  aftersaleInfo?: string
  logistics?: string
  logisticsNo: string
  carrier?: string
  returnLocation: string
  shipTime?: string
  applyTime?: string
  returnTime?: string
  syncedAt: string
}

export interface ShippedRefund {
  id: number
  shopId: number
  shopName: string
  platformAftersaleId: string
  orderNo: string
  productTitle: string
  productImage?: string
  sku: string
  productTags?: string
  tags?: string
  qty: number
  buyQty?: number
  payAmount: string
  refundAmount: string
  aftersaleType: string
  reason: string
  status: string
  orderInfo?: string
  aftersaleInfo?: string
  logistics?: string
  logisticsStatus?: string
  logisticsNo?: string
  carrier?: string
  shipTime?: string
  tracks?: LogisticsTrack[]
  alert: boolean
  applyTime?: string
  syncedAt: string
}

export interface LogisticsTrack {
  date?: string
  title?: string
  detail?: string
  text?: string
}

export async function fetchReturnPackages(params?: {
  shopId?: number
  keyword?: string
  returnFrom?: string
  returnTo?: string
  applyFrom?: string
  applyTo?: string
  page?: number
  pageSize?: number
}) {
  return unwrap<PageData<ReturnPackage>>(await client.get('/return-packages', { params }))
}

export async function fetchShippedRefunds(params?: {
  shopId?: number
  keyword?: string
  status?: string
  alertOnly?: boolean
  applyFrom?: string
  applyTo?: string
  page?: number
  pageSize?: number
}) {
  return unwrap<PageData<ShippedRefund>>(await client.get('/shipped-refunds', {
    params: {
      ...params,
      alertOnly: params?.alertOnly ? '1' : undefined,
    },
  }))
}

export interface InterceptOrder {
  id: number
  shopId: number
  shopName: string
  source: string
  needIntercept: boolean
  awaitPickup: boolean
  platformAftersaleId: string
  orderNo: string
  productTitle: string
  productImage?: string
  sku: string
  qty: number
  buyQty?: number
  payAmount: string
  refundAmount: string
  aftersaleType: string
  reason: string
  status: string
  logistics?: string
  logisticsStatus?: string
  logisticsNo?: string
  shipLogisticsNo?: string
  returnLogisticsNo?: string
  carrier?: string
  applyTime?: string
  syncedAt: string
}

export async function fetchInterceptOrders(params?: {
  shopId?: number
  keyword?: string
  page?: number
  pageSize?: number
}) {
  return unwrap<PageData<InterceptOrder>>(await client.get('/intercept-orders', { params }))
}

File: internal/dto/shop.go
package dto

type ShopCreateInput struct {
	Name     string `json:"name" binding:"required"`
	Platform string `json:"platform"`
	Remark   string `json:"remark"`
}

type ShopUpdateInput struct {
	Name   string `json:"name"`
	Remark string `json:"remark"`
}

type ShopItem struct {
	ID                 uint64 `json:"id"`
	Name               string `json:"name"`
	Platform           string `json:"platform"`
	PlatformLabel      string `json:"platformLabel"`
	BindCode           string `json:"bindCode"`
	PluginStatus       string `json:"pluginStatus"`
	PluginAvailable    bool   `json:"pluginAvailable"`
	PlatformShopID     string `json:"platformShopId,omitempty"`
	PlatformShopName   string `json:"platformShopName,omitempty"`
	LastSyncAt         string `json:"lastSyncAt,omitempty"`
	LastSeenAt         string `json:"lastSeenAt,omitempty"`
	NextSyncAt         string `json:"nextSyncAt,omitempty"`
	SyncRequested      bool   `json:"syncRequested"`
	PendingTicketCount int    `json:"pendingTicketCount,omitempty"`
	Remark             string `json:"remark,omitempty"`
	CreatedAt          string `json:"createdAt"`
	UpdatedAt          string `json:"updatedAt"`
}

type PluginHeartbeatResult struct {
	ShopItem
	SyncNow         bool `json:"syncNow"`
	SyncIntervalSec int  `json:"syncIntervalSec"`
}

type PluginSetting struct {
	PluginSyncIntervalMin int `json:"pluginSyncIntervalMin"`
}

type FilterCardItem struct {
	GroupName string `json:"groupName"`
	CardKey   string `json:"cardKey"`
	CardLabel string `json:"cardLabel"`
	Count     int    `json:"count"`
	SortOrder int    `json:"sortOrder"`
}

type TicketItem struct {
	ID                   uint64   `json:"id"`
	PlatformAftersaleID  string   `json:"platformAftersaleId"`
	OrderNo              string   `json:"orderNo"`
	ProductTitle         string   `json:"productTitle"`
	ProductImage         string   `json:"productImage,omitempty"`
	SKU                  string   `json:"sku"`
	ProductTags          string   `json:"productTags,omitempty"`
	Tags                 string   `json:"tags,omitempty"`
	Qty                  int      `json:"qty"`
	BuyQty               int      `json:"buyQty"`
	PayAmount            string   `json:"payAmount"`
	RefundAmount         string   `json:"refundAmount"`
	AftersaleType        string   `json:"aftersaleType"`
	Reason               string   `json:"reason"`
	Status               string   `json:"status"`
	TimeoutText          string   `json:"timeoutText,omitempty"`
	TimeoutAction        string   `json:"timeoutAction,omitempty"`
	DeadlineAt           string   `json:"deadlineAt,omitempty"`
	RemainSeconds        int      `json:"remainSeconds"`
	Dispute              string   `json:"dispute,omitempty"`
	Logistics            string   `json:"logistics,omitempty"`
	LogisticsBuyerStatus string   `json:"logisticsBuyerStatus,omitempty"`
	LogisticsShipStatus  string   `json:"logisticsShipStatus,omitempty"`
	NeedIntercept        bool     `json:"needIntercept,omitempty"`
	ReturnLogisticsNo    string   `json:"returnLogisticsNo,omitempty"`
	ShipLogisticsNo      string   `json:"shipLogisticsNo,omitempty"`
	ApplyTime            string   `json:"applyTime,omitempty"`
	CardKeys             []string `json:"cardKeys"`
	SyncedAt             string   `json:"syncedAt"`
}

type ShopWorkbench struct {
	Shop       ShopItem         `json:"shop"`
	Cards      []FilterCardItem `json:"cards"`
	Tickets    []TicketItem     `json:"tickets"`
	Total      int64            `json:"total"`
	Page       int              `json:"page"`
	PageSize   int              `json:"pageSize"`
	LastSyncAt string           `json:"lastSyncAt,omitempty"`
}

type PluginBindInput struct {
	BindCode string `json:"bindCode" binding:"required"`
}

type PluginBindResult struct {
	ShopID       uint64 `json:"shopId"`
	ShopName     string `json:"shopName"`
	Platform     string `json:"platform"`
	PluginKey    string `json:"pluginKey"`
	PluginSecret string `json:"pluginSecret"`
}

type PluginAuthInput struct {
	PluginKey    string `json:"pluginKey"`
	PluginSecret string `json:"pluginSecret"`
}

type PluginHeartbeatInput struct {
	PluginAuthInput
	PlatformShopID   string `json:"platformShopId"`
	PlatformShopName string `json:"platformShopName"`
}

type PluginSyncCard struct {
	GroupName string `json:"groupName"`
	CardKey   string `json:"cardKey"`
	CardLabel string `json:"cardLabel"`
	Count     int    `json:"count"`
	SortOrder int    `json:"sortOrder"`
}

type PluginSyncTicket struct {
	PlatformAftersaleID string   `json:"platformAftersaleId"`
	OrderNo             string   `json:"orderNo"`
	ProductTitle        string   `json:"productTitle"`
	ProductImage        string   `json:"productImage"`
	SKU                 string   `json:"sku"`
	ProductTags         string   `json:"productTags"`
	Tags                string   `json:"tags"`
	Qty                 int      `json:"qty"`
	BuyQty              int      `json:"buyQty"`
	PayAmount           string   `json:"payAmount"`
	RefundAmount        string   `json:"refundAmount"`
	AftersaleType       string   `json:"aftersaleType"`
	Reason              string   `json:"reason"`
	Status              string   `json:"status"`
	TimeoutText         string   `json:"timeoutText"`
	Dispute             string   `json:"dispute"`
	Logistics           string   `json:"logistics"`
	ReturnLogisticsNo   string   `json:"returnLogisticsNo"`
	ShipLogisticsNo     string   `json:"shipLogisticsNo"`
	ApplyTime           string   `json:"applyTime"`
	CardKeys            []string `json:"cardKeys"`
	RawJSON             string   `json:"rawJson"`
}

type PluginSyncReturn struct {
	PlatformAftersaleID string `json:"platformAftersaleId"`
	OrderNo             string `json:"orderNo"`
	ProductTitle        string `json:"productTitle"`
	ProductImage        string `json:"productImage"`
	SKU                 string `json:"sku"`
	Qty                 int    `json:"qty"`
	BuyQty              int    `json:"buyQty"`
	PayAmount           string `json:"payAmount"`
	RefundAmount        string `json:"refundAmount"`
	AftersaleType       string `json:"aftersaleType"`
	Reason              string `json:"reason"`
	Status              string `json:"status"`
	OrderInfo           string `json:"orderInfo"`
	AftersaleInfo       string `json:"aftersaleInfo"`
	Logistics           string `json:"logistics"`
	LogisticsNo         string `json:"logisticsNo"`
	Carrier             string `json:"carrier"`
	ReturnLocation      string `json:"returnLocation"`
	ShipTime            string `json:"shipTime"`
	ApplyTime           string `json:"applyTime"`
	ReturnTime          string `json:"returnTime"`
	TrackJSON           string `json:"trackJson"`
	RawJSON             string `json:"rawJson"`
}

type PluginSyncInput struct {
	PluginAuthInput
	PlatformShopID   string                     `json:"platformShopId"`
	PlatformShopName string                     `json:"platformShopName"`
	Cards            []PluginSyncCard           `json:"cards"`
	Tickets          []PluginSyncTicket         `json:"tickets"`
	Returns          *[]PluginSyncReturn        `json:"returns"`
	ShippedRefunds   *[]PluginSyncShippedRefund `json:"shippedRefunds"`
	ServiceOrders    *[]PluginSyncServiceOrder  `json:"serviceOrders"`
}

type PluginSyncServiceOrder struct {
	PlatformServiceID string `json:"platformServiceId"`
	OrderNo           string `json:"orderNo"`
	ProductTitle      string `json:"productTitle"`
	ProductImage      string `json:"productImage"`
	ProductContent    string `json:"productContent"`
	BuyerNick         string `json:"buyerNick"`
	CreateSource      string `json:"createSource"`
	BusinessType      string `json:"businessType"`
	OrderType         string `json:"orderType"`
	Tags              string `json:"tags"`
	StatusTab         string `json:"statusTab"`
	Status            string `json:"status"`
	TimeoutText       string `json:"timeoutText"`
	DelayEndTime      int64  `json:"delayEndTime"`
	DelayTimeLeft     int64  `json:"delayTimeLeft"`
	Detail            string `json:"detail"`
	Solution          string `json:"solution"`
	LastLog           string `json:"lastLog"`
	LastLogTime       string `json:"lastLogTime"`
	CreateTime        string `json:"createTime"`
	RawJSON           string `json:"rawJson"`
}

type PluginSyncShippedRefund struct {
	PlatformAftersaleID string `json:"platformAftersaleId"`
	OrderNo             string `json:"orderNo"`
	ProductTitle        string `json:"productTitle"`
	ProductImage        string `json:"productImage"`
	SKU                 string `json:"sku"`
	ProductTags         string `json:"productTags"`
	Tags                string `json:"tags"`
	Qty                 int    `json:"qty"`
	BuyQty              int    `json:"buyQty"`
	PayAmount           string `json:"payAmount"`
	RefundAmount        string `json:"refundAmount"`
	AftersaleType       string `json:"aftersaleType"`
	Reason              string `json:"reason"`
	Status              string `json:"status"`
	OrderInfo           string `json:"orderInfo"`
	AftersaleInfo       string `json:"aftersaleInfo"`
	Logistics           string `json:"logistics"`
	LogisticsStatus     string `json:"logisticsStatus"`
	LogisticsNo         string `json:"logisticsNo"`
	Carrier             string `json:"carrier"`
	ShipTime            string `json:"shipTime"`
	TrackJSON           string `json:"trackJson"`
	ApplyTime           string `json:"applyTime"`
	RawJSON             string `json:"rawJson"`
}

type PluginSyncResult struct {
	ShopID             uint64 `json:"shopId"`
	CardCount          int    `json:"cardCount"`
	TicketCount        int    `json:"ticketCount"`
	ReturnCount        int    `json:"returnCount"`
	ShippedRefundCount int    `json:"shippedRefundCount"`
	ServiceOrderCount  int    `json:"serviceOrderCount"`
	LastSyncAt         string `json:"lastSyncAt"`
}

type ServiceOrderItem struct {
	ID                uint64 `json:"id"`
	ShopID            uint64 `json:"shopId"`
	ShopName          string `json:"shopName"`
	PlatformServiceID string `json:"platformServiceId"`
	OrderNo           string `json:"orderNo"`
	ProductTitle      string `json:"productTitle"`
	ProductImage      string `json:"productImage,omitempty"`
	ProductContent    string `json:"productContent,omitempty"`
	BuyerNick         string `json:"buyerNick,omitempty"`
	CreateSource      string `json:"createSource,omitempty"`
	BusinessType      string `json:"businessType,omitempty"`
	OrderType         string `json:"orderType,omitempty"`
	Tags              string `json:"tags,omitempty"`
	StatusTab         string `json:"statusTab"`
	Status            string `json:"status"`
	TimeoutText       string `json:"timeoutText,omitempty"`
	TimeoutAction     string `json:"timeoutAction,omitempty"`
	DeadlineAt        string `json:"deadlineAt,omitempty"`
	RemainSeconds     int    `json:"remainSeconds"`
	Detail            string `json:"detail,omitempty"`
	Solution          string `json:"solution,omitempty"`
	LastLog           string `json:"lastLog,omitempty"`
	LastLogTime       string `json:"lastLogTime,omitempty"`
	CreateTime        string `json:"createTime,omitempty"`
	SyncedAt          string `json:"syncedAt"`
}

type ServiceTabCount struct {
	StatusTab string `json:"statusTab"`
	Count     int64  `json:"count"`
}

type ServiceOrderListQuery struct {
	ShopID    uint64
	StatusTab string
	Keyword   string
	Page      int
	PageSize  int
}

type ServiceOrderListResult struct {
	List     []ServiceOrderItem `json:"list"`
	Total    int64              `json:"total"`
	Page     int                `json:"page"`
	PageSize int                `json:"pageSize"`
	Tabs     []ServiceTabCount  `json:"tabs"`
}

type ReturnPackageItem struct {
	ID                  uint64 `json:"id"`
	ShopID              uint64 `json:"shopId"`
	ShopName            string `json:"shopName"`
	PlatformAftersaleID string `json:"platformAftersaleId"`
	OrderNo             string `json:"orderNo"`
	ProductTitle        string `json:"productTitle"`
	ProductImage        string `json:"productImage,omitempty"`
	SKU                 string `json:"sku"`
	Qty                 int    `json:"qty"`
	BuyQty              int    `json:"buyQty"`
	PayAmount           string `json:"payAmount"`
	RefundAmount        string `json:"refundAmount"`
	AftersaleType       string `json:"aftersaleType"`
	Reason              string `json:"reason"`
	Status              string `json:"status"`
	OrderInfo           string `json:"orderInfo,omitempty"`
	AftersaleInfo       string `json:"aftersaleInfo,omitempty"`
	Logistics           string `json:"logistics,omitempty"`
	LogisticsNo         string `json:"logisticsNo"`
	Carrier             string `json:"carrier,omitempty"`
	ReturnLocation      string `json:"returnLocation"`
	ShipTime            string `json:"shipTime,omitempty"`
	ApplyTime           string `json:"applyTime,omitempty"`
	ReturnTime          string `json:"returnTime,omitempty"`
	SyncedAt            string `json:"syncedAt"`
}

type ReturnListQuery struct {
	ShopID     uint64
	Keyword    string
	ReturnFrom string
	ReturnTo   string
	ApplyFrom  string
	ApplyTo    string
	Page       int
	PageSize   int
}

type ShippedRefundItem struct {
	ID                  uint64           `json:"id"`
	ShopID              uint64           `json:"shopId"`
	ShopName            string           `json:"shopName"`
	PlatformAftersaleID string           `json:"platformAftersaleId"`
	OrderNo             string           `json:"orderNo"`
	ProductTitle        string           `json:"productTitle"`
	ProductImage        string           `json:"productImage,omitempty"`
	SKU                 string           `json:"sku"`
	ProductTags         string           `json:"productTags,omitempty"`
	Tags                string           `json:"tags,omitempty"`
	Qty                 int              `json:"qty"`
	BuyQty              int              `json:"buyQty"`
	PayAmount           string           `json:"payAmount"`
	RefundAmount        string           `json:"refundAmount"`
	AftersaleType       string           `json:"aftersaleType"`
	Reason              string           `json:"reason"`
	Status              string           `json:"status"`
	OrderInfo           string           `json:"orderInfo,omitempty"`
	AftersaleInfo       string           `json:"aftersaleInfo,omitempty"`
	Logistics           string           `json:"logistics,omitempty"`
	LogisticsStatus     string           `json:"logisticsStatus,omitempty"`
	LogisticsNo         string           `json:"logisticsNo,omitempty"`
	Carrier             string           `json:"carrier,omitempty"`
	ShipTime            string           `json:"shipTime,omitempty"`
	Tracks              []LogisticsTrack `json:"tracks,omitempty"`
	Alert               bool             `json:"alert"`
	ApplyTime           string           `json:"applyTime,omitempty"`
	SyncedAt            string           `json:"syncedAt"`
}

type LogisticsTrack struct {
	Date   string `json:"date"`
	Title  string `json:"title"`
	Detail string `json:"detail"`
	Text   string `json:"text"`
}

type ShippedRefundListQuery struct {
	ShopID    uint64
	Keyword   string
	Status    string
	AlertOnly bool
	ApplyFrom string
	ApplyTo   string
	Page      int
	PageSize  int
}

type InterceptItem struct {
	ID                  uint64 `json:"id"`
	ShopID              uint64 `json:"shopId"`
	ShopName            string `json:"shopName"`
	Source              string `json:"source"`
	NeedIntercept       bool   `json:"needIntercept"`
	AwaitPickup         bool   `json:"awaitPickup"`
	PlatformAftersaleID string `json:"platformAftersaleId"`
	OrderNo             string `json:"orderNo"`
	ProductTitle        string `json:"productTitle"`
	ProductImage        string `json:"productImage,omitempty"`
	SKU                 string `json:"sku"`
	Qty                 int    `json:"qty"`
	BuyQty              int    `json:"buyQty"`
	PayAmount           string `json:"payAmount"`
	RefundAmount        string `json:"refundAmount"`
	AftersaleType       string `json:"aftersaleType"`
	Reason              string `json:"reason"`
	Status              string `json:"status"`
	Logistics           string `json:"logistics,omitempty"`
	LogisticsStatus     string `json:"logisticsStatus,omitempty"`
	LogisticsNo         string `json:"logisticsNo,omitempty"`
	ShipLogisticsNo     string `json:"shipLogisticsNo,omitempty"`
	ReturnLogisticsNo   string `json:"returnLogisticsNo,omitempty"`
	Carrier             string `json:"carrier,omitempty"`
	ApplyTime           string `json:"applyTime,omitempty"`
	SyncedAt            string `json:"syncedAt"`
}

type InterceptListQuery struct {
	ShopID   uint64
	Keyword  string
	Page     int
	PageSize int
}
